python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -1769,25 +1766,25 @@ def computeAltMaskList(self, tempExpRefList, maskSpanSets):
 
         return altMaskList
 
-    def _filterArtifacts(self, spanSetList, epochCountImage, maxNumEpochs=None):
+    def _filterArtifacts(self, spanSetList, epochCountImage):
         maskSpanSetList = []
         x0, y0 = epochCountImage.getXY0()
         for i, span in enumerate(spanSetList):
             y, x = span.indices()
             counts = epochCountImage.array[[y1 - y0 for y1 in y], [x1 - x0 for x1 in x]]
-            idx = numpy.where((counts > 0) & (counts <= maxNumEpochs))
+            idx = numpy.where((counts > 0) & (counts <= self.config.maxNumEpochs))
             percentBelowThreshold = len(idx[0]) / len(counts)
             if percentBelowThreshold > self.config.spatialThreshold:
                 maskSpanSetList.append(span)
         return maskSpanSetList
 
     def _readAndComputeWarpDiff(self, warpRef, imageScaler, templateCoadd):
-        # Warp comparison must use PSF-Matched Warps regardless of requested coadd warp type
-        warpName = self.getTempExpDatasetName('psfMatched')
+        # Warp comparison should use PSF-Matched Warps regardless of requested coadd warp type
+        warpName = self.getTempExpDatasetName(self.config.assembleStaticSkyModel.warpType)
```

Comment:
It's not a good idea to go reaching through multiple config elements to get what you want: what if I `retarget` the `assembleStaticSkyModel` and name `warpType` something else? You should only access what you control. That suggests putting `warpType` into the config at _this_ level, and passing it down to the `assembleStaticSkyModel` as it needs it.

Reply:
Hmmm. How would you have done this?  Passed the new config down to the subtask in `validate`?  I configure the defaults for assembleStaticSkyModel subtask (an `AssembleCoaddTask` instance) in the CompareWarpAssembleConfig's `setDefaults`, already requiring that it have a warpType. It's going to be 'psfMatched' in every scenario I can think of (but then again I lack imagination).  I'm debating if its worth making a new config parameter or just leaving it hardcoded.

Reply:
We could just leave it hardcoded for now: I also lack the imagination to think as to why anyone would use a different `warpType`, since the algorithm for rejecting artifacts requires PSF-matched inputs.
Or put `warpType` into the `AssembleCoaddTask`'s `Config`, and then pass it to the `AssembleStaticSkyModelTask` constructor when you `makeSubtask`.

Reply:
I thought configs frozen by the time we got to making the subtask, but I'll give it try.

Reply:
The config is definitely frozen within `CompareWarpAssembleCoadd.__init__`

```
(Pdb) kwargs['config'].assembleStaticSkyModel.warpType = staticSkyModelWarpType
*** FieldValidationError: ConfigurableField 'assembleStaticSkyModel' failed validation: Cannot modify a frozen Config
For more information read the Field definition at:
  File pex/config/configurableField.py:271 (__deepcopy__)And the Config definition at:
  File pipe/tasks/assembleCoadd.py:1439 (<module>)
```
To clarify: `AssembleStaticSkyModelTask` is not a thing.   It is a vanilla `AssembleCoaddTask` used as a subtask within `CompareWarpAssembleCoaddTask`. 

Only solution I can think of at the moment is:
```    
def validate(self):
        AssembleCoaddConfig.validate(self)
        self.assembleStaticSkyModel.warpType = self.staticSkyModelWarpType
```
which I'm not sure is any better because it smells like code duplication. `staticSkyModelWarpType` is just a copied and pasted `warpType` config from:
https://github.com/lsst/pipe_tasks/blob/master/python/lsst/pipe/tasks/assembleCoadd.py#L55

Reply:
Reverting to hardcoded 'psfMatched'. Jenkins running.